submissions/BlueLamantine/js-dom/dom.js

```diff
@@ -0,0 +1,144 @@
+	title: 'Americano',
+	id: 'americano',
+	classes: {
+		bgColor: 'americano-bg',
+		textColor: 'cappuccino-text'
+	},
+	paragraph: 'Shots of espresso diluted with water.',
+}, {
+	title: 'Macchiato',
+	id: 'macchiato',
+	classes: {
+		bgColor: 'macchiato-bg',
+		textColor: 'cappuccino-text'
+	},
+	paragraph: 'Espresso with a dash of frothy foamed milk.',
+}, {
+	title: 'Flat white',
+	id: 'flatw',
+	classes: {
+		bgColor: 'flatw-bg',
+		textColor: 'americano-text'
+	},
+	paragraph: 'Double shot of espresso with steamed milk.',
+}, {
+	title: 'Cappuccino',
+	id: 'cappuccino',
+	classes: {
+		bgColor: 'cappuccino-bg',
+		textColor: 'americano-text'
+	},
+	paragraph: 'Double shot of espresso topped with an airy thick layer of foamed milk.',
+}, {
+	title: 'Caffe latte',
+	id: 'latte',
+	classes: {
+		bgColor: 'latte-bg',
+		textColor: 'doppio-text'
+	},
+	paragraph: 'A shot of espresso in steamed milk lightly topped with foam.',
+}];
+const MENU = document.getElementById('mainMenu');
+const CUP = document.getElementById('coffeeCups');
+const ART = document.getElementById('articles');
+
+function render() {
+	COFFEE.forEach(element => {
+		MENU.appendChild(createListNode(element));
+		CUP.appendChild(createCupNode(element));
+		ART.appendChild(createContentNode(element));
+	});
+};
+
+function createListNode(el) {
+	let listItem = document.createElement('li');
+	listItem.className = `menu-item ${el.classes.bgColor}`;
+	listItem.innerHTML = `
+    <a href="#" id="${el.id}" class="item__link ${el.classes.textColor}">${el.title}</a>
+    `;
+	if (el.id == 'doppio') {
+		listItem.classList.add('active');
+	}
+	return listItem;
+};
+
+function createCupNode(el) {
+	let cup = document.createElement('div');
+	cup.id = `${el.id}-cup`;
+	cup.className = `cup ${el.id}-cup__bg `;
+	if (el.id !== 'doppio') {
+		cup.hidden = true;
+	}
+	cup.innerHTML = `
+        <div class = "paw-left"></div>
+        <div class = "paw-right"></div>
+    `;
+	return cup;
+};
+
+function createContentNode(el) {
+	const articleContent = document.createElement('article');
+	const headerContent = document.createElement('h2');
+	const textContent = document.createElement('p');
+	articleContent.id = `${el.id}-article`;
+	articleContent.className = 'description';
+	if (el.id !== 'doppio') {
+		articleContent.hidden = true;
+	}
+	headerContent.innerHTML = el.title;
+	textContent.innerHTML = el.paragraph;
+	articleContent.appendChild(headerContent);
+	articleContent.appendChild(textContent);
+	return articleContent;
+};
+
+function toggleMenuItem() {
+	const content = {
+		articles: [...document.getElementsByTagName('article')],
+		cups: [...document.getElementsByClassName('cup')]
+	}
+	MENU.addEventListener('click', e => {
+		e.preventDefault();
+		if (e.target.matches('.item__link')) {
+			const arrLi = [...MENU.children];
+			const isActive = arrLi.find(li => li.classList.contains("active"));
+			isActive.classList.remove('active');
+			e.target.parentNode.classList.add('active');
```

Comment:
Be attentive with using `e.target.parentNode` because it strongly depends on your layout. It's better to use `closest`. More info [here](https://developer.mozilla.org/ru/docs/Web/API/Element/closest)

Reply:
Ok, thanks for a good tip!